### DISCUSSION TITLE
Allow non-table to be seen as tables

### DISCUSSION THREAD

**Discussion by darioguarascio** [directus/directus#12673] [Feature Requests]
Hello,

I ended up in an edge case where I would like to use [postgres_fdw](https://www.postgresql.org/docs/current/postgres-fdw.html) in order to share some data across different dbs.

I tried the following approach:
- create a table in directus
- access `/content/table`
- drop the table in db
- import foreign table in db
- refresh `/content/table`

All looks good, table is usable, readable. the fact it is a foreing table is completely transparent.

If i navigate to settings tho, the table is then marked as a folder (I suspect some internal checks about schemas onyl allowing strictly typed tables to be seen as tables)

The same would be valid for views, materialized views and so on.

Is there any chance to support them, even if read-only mode ?

**Comment by rijkvanzanten** [comment]
> Is there any chance to support them, even if read-only mode ?

This is easier said then done.. The schema introspection needs to be built in a way to treat those tables the correct way, rather than just treating them as regular tables (as that will cause other issues). Support for SQL Views is something we're very interested in supporting (also see https://github.com/directus/directus/discussions/3097), but I can't make any calls on `postgres_fdw`, as that would be a bespoke implementation for just postgres 🤔 (hopefully those can be treated the same as views)

**Reply by darioguarascio** [reply]
I guess building a view out of a table made with this approach would work as well :+1:

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
